examples/indexing/fancy_indexing.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from numpy.random import randint
+
+# %out = tensor.gather %input[%indices] gather_along_dimension(1) :
+#   tensor<2x3x4xf32>[tensor<5x6x 1xindex>] -> tensor<5x6x2x1x4xf32>
+
+input = randint(0, 10, (2, 3, 4))
+indices = randint(0, 2, (5, 6))
+out = input[:, indices, :]
+out_ = np.moveaxis(out, (1, 2), (0, 1))
+
+for index in np.ndindex(5, 6):
+  coord = indices[index]
+  assert np.allclose(out_[index], input[:, coord, :])
+
+# %out = tensor.gather %input[%indices] gather_along_dimensions(0, 2) :
+#   (tensor<2x3x4xf32>, tensor<5x6x 2xindex>) -> tensor<5x6x3xf32>
+
+indices = randint(0, 1, (5, 6, 2))
```

Comment:
Again, I am still wrapping my head around the basics here. But `randint(0, 1, ...)` produces only zeroes... is that intended? It also seems like in the next line, the indices used for the first axis of `input` can be from `[0..1]` whereas those used for the third axis can be from `[0..3]`, so even if we change the line above to `randint(0, 2, ...)`, there still seems to be something off because that wouldn't ever read the whole input. Is that intended?

Reply:
Ah whoops I thought it was `[low, high]` but you're right it's `[low, high)`.

>the indices used for the first axis of input can be from [0..1] whereas those used for the third axis can be from [0..3]

Yes your intuition is right on - the `indices` tensor should be exercising indexing anywhere along the 0th dim and 2nd dim (where the extent of the 2nd dim is `[0..3]`).  I just didn't want to have some verbose numpy line that created an `indices` array that was something like `5 x 6 x randint(0,2) x randint(0,4)`. But you're right I should change it so that all the numbers line up intuitively.

Reply:
Thanks for clarifying/confirming! As I said, I still working on understanding the basics and I asked these questions because I these subtleties might, in fact, be me fundamentally misunderstanding what is going on... BTW, this file was hugely helpful for me! Thanks :)